File: apps/backend/app/contracts/aiExecutionPreferenceContract.py
from typing import Literal

from pydantic import BaseModel


class AiExecutionPreferenceContract(BaseModel):
    contractName: Literal[
        "AiExecutionPreference.v1"
    ] = "AiExecutionPreference.v1"

    mode: Literal["AUTO", "FIXED", "CUSTOM"] = "AUTO"
    fixedModelId: str | None = None
    textModelId: str | None = None
    visionModelId: str | None = None
    embeddingModelId: str | None = None
    thinkingMode: Literal["AUTO", "ON", "OFF"] = "AUTO"


class MaterialAiPreferenceUpdateContract(
    AiExecutionPreferenceContract
):
    contractName: Literal[
        "MaterialAiPreferenceUpdate.v1"
    ] = "MaterialAiPreferenceUpdate.v1"

File: Documentacao/LIA2_008_PedagogicalAdaptive_Mobile_Thinking_FIX02/apps/studentWeb/src/contracts/aiExecutionPreferenceContract.ts
export type AiExecutionMode = 'AUTO' | 'FIXED' | 'CUSTOM'
export type ThinkingMode = 'AUTO' | 'ON' | 'OFF'

export interface AiExecutionPreferenceContract {
  contractName: 'AiExecutionPreference.v1'
  mode: AiExecutionMode
  fixedModelId: string | null
  textModelId: string | null
  visionModelId: string | null
  embeddingModelId: string | null
  thinkingMode: ThinkingMode
}
